core/tests/protractor_utils/TopicViewerPage.js

```diff
@@ -58,6 +60,35 @@ var TopicViewerPage = function() {
       expect(await storySummaryTitleList.count()).toEqual(count);
     }
   };
+
+  this.moveToRevisionTab = async function() {
+    var revisionTabLink = element(by.css('.protractor-test-revision-tab-link'));
+    await action.click('Revision Tab', revisionTabLink);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.moveToPracticeTab = async function() {
+    var practiceTabLink = element(by.css('.protractor-test-practice-tab-link'));
+    await action.click('Practice Tab', practiceTabLink);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.selectSkillForPractice = async function(subtopicTitle) {
+    var skillCheckbox = element(by.cssContainingText(
+      '.protractor-test-skill-checkbox', subtopicTitle));
+    await action.click('Select skill to practice', skillCheckbox);
+  };
+
+  this.startPractice = async function() {
+    await action.click('Start practice', startPracticeButton);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.expectedMessageAfterCompletion = async function(message) {
+    var messageOnCompletion = element(
+      by.css('.protractor-test-practice-complete-message'));
+    expect(await messageOnCompletion.getText()).toEqual(message);
```

Comment:
Can you wait for the `messageOnCompletion` element to be visible before calling `getText` on it?

Reply:
Done